graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/execution/FederatedTypeRegistry.kt

```diff
@@ -19,10 +19,15 @@ package com.expediagroup.graphql.federation.execution
 /**
  * Simple registry that holds mapping of all registered federated GraphQL types and their corresponding resolvers.
  */
-class FederatedTypeRegistry(private val federatedTypeResolvers: Map<String, FederatedTypeResolver<*>> = emptyMap()) {
+class FederatedTypeRegistry(federatedTypeResolvers: List<FederatedTypeResolver<*>> = emptyList()) {
+
+    /**
+     * Precompute the map of names to types so that we don't have to search the list every time.
+     */
+    private val namesToResolvers: Map<String, FederatedTypeResolver<*>> = federatedTypeResolvers.map { it.typeName to it }.toMap()
```

Comment:
instead of `map { }.toMap()` I think you can just use `associateBy { it.typeName }`